File: app/src/main/java/com/applications/whazzup/photomapp/di/DaggerService.java
package com.applications.whazzup.photomapp.di;


import android.content.Context;

public class DaggerService {
    public static final String SERVICE_NAME = "SERVICE_NAME";

    @SuppressWarnings("unchecked")
    public static <T> T getDaggerComponent(Context context) {
        //noinspection ResourceType
        return (T) context.getSystemService(SERVICE_NAME);
    }
}

File: app/src/main/java/com/applications/whazzup/photomapp/di/DaggerService.kt
package com.applications.whazzup.photomapp.di

import android.content.Context


class DaggerService {

    companion object {
        val SERVICE_NAME = "SERVICE_NAME"

        fun <T> getDaggerComponent(context: Context): T {
            @Suppress("UNCHECKED_CAST")
            return context.getSystemService(SERVICE_NAME) as T
        }
    }
}
